airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -145,20 +165,26 @@ public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws Retr
 
               final StandardSyncSummary standardSyncSummary = standardSyncOutput.get().getStandardSyncSummary();
 
+              // if the current workflow was a reset, it is now safe to switch it back to "normal".
               if (workflowState.isResetConnection()) {
                 workflowState.setResetConnection(false);
               }
 
               if (standardSyncSummary != null && standardSyncSummary.getStatus() == ReplicationStatus.FAILED) {
+                // todo (cgardens) - why are failures a member field as opposed to a local variable? seems like it
+                // is just asking for us to make a mistake when resetting it.
```

Comment:
We should discuss that with @pmossman , I am not sure to have all the relevant information about that.

Reply:
I had trouble coming up with an elegant way to use a local variable here because all of this logic is happening inside the cancellation scope lambda. Good point that this is error prone though, I'll revisit and see if I can come up with something